metrics-operator/controllers/common/providers/prometheus/common.go

```diff
@@ -40,13 +41,21 @@ func (r RoundTripperRetriever) GetRoundTripper(ctx context.Context, provider met
 		}
 		return nil, err
 	}
-	return config.NewBasicAuthRoundTripper(secret.User, secret.Password, "", "", promapi.DefaultRoundTripper), nil
+
+	transport := &http.Transport{
```

Comment:
looks good, but please also add the parameters and values from the `DefaultRoundTripper` so we won't break any existing behavior